### PR TITLE
time: use boot time for timer instead of real time (bsc#1129986)

### PR DESCRIPTION
**Pull request by mtomaschewski**
Fixes to use boot time for timers/timeouts and real time only in lease
files to not stuck in timers due to adjustments of the system time.
Consistently log dhcp xid and enabled to log dhcp6 timings line.

### CODE REVIEW

File: src/wpa-supplicant.c [hunk 1]
```diff
@@ -622,7 +627,7 @@ ni_wpa_interface_retrieve_scan(ni_wpa_interface_t *wpa_dev, ni_wireless_scan_t *
 	if (ni_wpa_interface_expire_networks(wpa_dev, scan->interval + 1) == 0) {
 		/* Nothing pruned. If we didn't receive new scan results in the
 		 * mean time, there's nothing we need to do. */
-		if (scan->timestamp == wpa_dev->scan.timestamp)
+		if (!timercmp(&scan->timestamp, &wpa_dev->scan.timestamp, !=))
```
Comment (rtorrero) [inline src/wpa-supplicant.c]: Is there any reason why you'd use a double negation here?

**Comment by rtorrero** [comment]
After some testing wtih tcpdump to see if DHCPv4 was doing renewal in time, everything seems to be working correctly with this PR. LGTM!

File: src/wpa-supplicant.c (hunk 1 above)
Comment (mtomaschewski) [reply src/wpa-supplicant.c]: See "man timercmp", there are some implementation that don't work properly with '==':
```Some  systems  (but  not Linux/glibc), have a broken timercmp() implementation, in which CMP of >=, <=, and == do not work; portable applications can instead use
           !timercmp(..., <)
           !timercmp(..., >)
           !timercmp(..., !=)
```

Comment (rtorrero) [reply src/wpa-supplicant.c]: You are totally right, marking as resolved

**Pull request merged**